src/node/index.d.ts

```diff
@@ -0,0 +1,526 @@
+     * @param serialize Serialization function for responses
+     * @param deserialize Deserialization function for requests
+     * @param type The streaming type of method that this handles
+     * @return True if the handler was set. False if a handler was already set for that name.
+     */
+    register(name: string, handler: () => void, serialize: (obj: any) => Buffer, deserialize: (buffer: Buffer) => any, type: string): boolean;
+
+    /**
+     * Add a service to the server, with a corresponding implementation. If you are
+     * generating this from a proto file, you should instead use `addProtoService`.
+     * @param service The service descriptor, as `getProtobufServiceAttrs` returns
+     * @param implementation Map of method names to method implementation for the provided service.
+     */
+    addService(service: { [index: string]: any }, implementation: { [index: string]: () => void }): void;
+
+    /**
+     * Add a proto service to the server, with a corresponding implementation
+     * @deprecated Use grpc.load and Server#addService instead
+     * @param service The proto service descriptor
+     * @param implementation Map of method names to method implementation for the provided service.
+     */
+    addProtoService(service: Service, implementation: any): void;
+
+    /**
+     * Binds the server to the given port, with SSL enabled if creds is given
+     * @param port The port that the server should bind on, in the format "address:port"
+     * @param creds Server credential object to be used for SSL. Pass an insecure credentials object for an insecure port.
+     */
+    bind(port: string, creds: ServerCredentials): any;
+  }
+
+  export interface IServerOptions {
+
+  }
+
+  export interface IError {
+    code: string;
+    message: string;
+  }
+
+  /**
+   * Credentials factories
+   */
+  export const credentials: {
+    /**
+     * Create an SSL Credentials object. If using a client-side certificate, both
+     * the second and third arguments must be passed.
+     * @param rootCerts The root certificate data
+     * @param privateKey The client certificate private key, if applicable
+     * @param certChain The client certificate cert chain, if applicable
+     * @return The SSL Credentials object
+     */
+    createSsl(rootCerts: Buffer, privateKey?: Buffer, certChain?: Buffer): ChannelCredentials;
+
+    /**
+     * Create a gRPC credentials object from a metadata generation function. This
+     * function gets the service URL and a callback as parameters. The error
+     * passed to the callback can optionally have a 'code' value attached to it,
+     * which corresponds to a status code that this library uses.
+     * @param metadataGenerator The function that generates metadata
+     * @return The credentials object
+     */
+    createFromMetadataGenerator(metadataGenerator: (s: string, callback: (error: Error, metadata: Metadata) => void) => void): CallCredentials;
```

Comment:
The metadata generator function actually takes a `{ service_url: string }` as its first argument. Also, the callback should be typed as `error: Error | null, metadata?: Metadata` to avoid compilation errors when using strict null checks 😄